src/averageAges.js

```diff
@@ -1,61 +1,52 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function arrayAverage(arr) {
+  const fullSum = arr.reduce((sum, a) => sum + a, 0);
+
+  return fullSum / arr.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const menOfTheCentury = !century ? men
+    : men.filter(person => Math.ceil(person.died / 100) === century);
+  const ages = menOfTheCentury.map(man => man.died - man.born);
+
+  return arrayAverage(ages);
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+  const filteredWomen = withChildren
+    ? women.filter(woman => people.some(person => person.mother === woman.name))
+    : women;
```

Comment:
could you try adding the condition directly to the women declaration?
this will help you avoid another filter

Reply:
@tetiana-ivanchenko Could you please help me to understand what I should do?

Reply:
like:
  const women = people.filter(woman => withChildren
    ? people.some(person => person.mother === woman.name)
    : people.filter(person => person.sex === 'f'));